packages/admin/admin/src/theme/componentsTheme/MuiDataGrid.tsx

```diff
@@ -146,6 +155,15 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
                 color: palette.grey[900],
             },
             alignContent: "center",
+
+            "&.MuiDataGrid-cell--pinnedLeft": {
```

Comment:
i think pinning on cell level is not an good idea, we shadow artifacts (shadow stops before the separator and starts again in the next cell) in between the cells:

<img width="826" alt="Screenshot 2025-06-30 at 13 11 50" src="https://github.com/user-attachments/assets/3dfdbdf6-547c-4590-acc4-fdc29b64dc9c" />



Futhermore i think we have a problem, when pinning more than 1 cell - we get shadows per cell:

![Screenshot 2025-06-30 at 13 12 54](https://github.com/user-attachments/assets/2ebec936-6844-4917-bdaf-cf7cbf7f192e)

How to pin multiple columns:
![Screen Recording 2025-06-30 at 13 12 42](https://github.com/user-attachments/assets/2105409c-8a66-446b-9b99-5daa7551749a)

Reply:
i just looked into it, i think you have to apply the styles when the cell/columnHeader has both classes set:

- .MuiDataGrid-cell--withLeftBorder / .MuiDataGrid-columnHeader--withLeftBorder
- .MuiDataGrid-cell--pinnedRight /   .MuiDataGrid-columnHeader--pinnedRight 


Following must be set to the columnHeader in my opinion
            [`&.${gridClasses["columnHeader--pinnedLeft"]}.${gridClasses["columnHeader--withRightBorder"]}`]: {
                boxShadow: theme.shadows[2],
            },

            [`&.${gridClasses["columnHeader--pinnedright"]}.${gridClasses["columnHeader--withLeftBorder"]}`]: {
                boxShadow: theme.shadows[2],
            },

and similar syntax for the cell.